Source/Core/Common/Arm64Emitter.h

```diff
@@ -474,18 +473,18 @@ class ArithOption
   {
     switch (m_type)
     {
-    case TYPE_EXTENDEDREG:
-      return (m_extend << 13) | (m_shift << 10);
-      break;
-    case TYPE_SHIFTEDREG:
+    case TypeSpecifier::ExtendedReg:
+      return (static_cast<u32>(m_extend) << 13) | (m_shift << 10);
+    case TypeSpecifier::ShiftedReg:
       return (m_shifttype << 22) | (m_shift << 10);
-      break;
     default:
       DEBUG_ASSERT_MSG(DYNA_REC, false, "Invalid type in GetData");
       break;
     }
     return 0;
   }
+
+  bool IsExtended() const { return GetType() == TypeSpecifier::ExtendedReg; }
```

Comment:
Perhaps remove ArithOption::GetType since TypeSpecifier is private.

Reply:
Ah, good catch. Removed it entirely, since it was just a simple getter